src/Controller/ActivityController.php

```diff
@@ -92,26 +115,16 @@ public function __construct(
      *     )
      * )
      */
-    public function getActivitiesList(ActivityRepository $repository): Response
+    public function getActivitiesList(int $page): JsonResponse
     {
         $user = $this->getUser();
-        $activities = $repository->getAvailableActivities($user);
-
-        /** @var SerializationContext $context */
-        $context = SerializationContext::create()->setGroups(array('ActivityList'));
-
-        $json = $this->serializer->serialize(
-            $activities,
-            'json',
-            $context
-        );
 
-        return new JsonResponse($json, 200, [], true);
+        return $this->activityHandler->getActivitiesListPaginated($user, $page);
     }
 
     /**
      * Get details about and Activity
-     * @Rest\Get("/{id}")
+     * @Rest\Get("/{id}", requirements={"id"="\d+"})
```

Comment:
Either `{page<\d+>}` or `requirements={"id"="\d+"}` should be used everywhere. Please change it everywhere you will find this.